src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -370,6 +370,8 @@ protected function createFormBuilder($data = null, array $options = []): FormBui
      * Shortcut to return the Doctrine Registry service.
      *
      * @throws \LogicException If DoctrineBundle is not available
+     *
+     * @deprecated since 5.4, inject an instance of ManagerRegistry in your controller instead
```

Comment:
what about a runtime deprecation as well? `deprecation-contracts` is available

Reply:
Done